promql/prettier/prettier_test.go

```diff
@@ -0,0 +1,446 @@
+		expr:     `quantile(0.9, sum(min(go_alloc_bytes) by (job)) without(job)) by (localhost)`,
+		expected: `quantile by (localhost) (0.9, sum without(job) (min by (job) (go_alloc_bytes)))`,
+	},
+	{
+		expr:     `quantile(0.9, sum(min(go_alloc_bytes) by (job)) without(job)) by (localhost)`,
+		expected: `quantile by(localhost) (0.9, sum without(job) (min by(job) (go_alloc_bytes)))`,
+	},
+	{
+		expr:     `quantile(0.9, sum(min(go_alloc_bytes) by (job)) ignoring(job)) by (localhost)`,
+		expected: `quantile by (localhost) (0.9, sum ignoring(job) (min by (job) (go_alloc_bytes)))`,
+	},
+	{
+		expr:     `quantile(0.9, sum(min(go_alloc_bytes) by (job)) ignoring(job)) by (localhost)`,
+		expected: `quantile by (localhost) (0.9, sum ignoring(job) (min by (job) (go_alloc_bytes)))`,
+	},
+	{
+		expr:     `sum(metric_first + metric_second) without(job)`,
+		expected: `sum without(job) (metric_first + metric_second)`,
+	},
+	{
+		expr:     `sum(metric_first + metric_second) without(job, instance, foo)`,
+		expected: `sum without(job, instance, foo) (metric_first + metric_second)`,
+	},
+	{
+		expr:     `min(sum(metric_first + metric_second) without(job, instance, foo)) by(job)`,
+		expected: `min by(job) (sum without(job, instance, foo) (metric_first + metric_second))`,
+	},
+	{
+		expr:     `min by(job) (sum(metric_first + metric_second) without(job, instance, foo))`,
+		expected: `min by(job) (sum without(job, instance, foo) (metric_first + metric_second))`,
+	},
+	{
+		expr:     `quantile(0.9, min by(job) (sum(metric_first + metric_second) without(job, instance, foo)))`,
+		expected: `quantile(0.9, min by(job) (sum without(job, instance, foo) (metric_first + metric_second)))`,
+	},
+	{
+		expr:     `quantile(0.9, min by(job) (sum(metric_first + metric_second) without(job, instance, foo))) ignoring(foo)`,
+		expected: `quantile ignoring(foo) (0.9, min by(job) (sum without(job, instance, foo) (metric_first + metric_second)))`,
+	},
+	{
+		expr:     `sum(go_alloc_bytes{job="prometheus", instance="localhost:9090"}) by (job)`,
+		expected: `sum by (job) (go_alloc_bytes{job="prometheus", instance="localhost:9090"})`,
+	},
+	{
+		expr:     `sum(go_alloc_bytes{job="prometheus", instance="localhost:9090"} + ignoring(instance) go_goroutines{job="prometheus", instance="localhost:9090"}) by (job)`,
+		expected: `sum by (job) (go_alloc_bytes{job="prometheus", instance="localhost:9090"} + ignoring(instance) go_goroutines{job="prometheus", instance="localhost:9090"})`,
+	},
+	// __name__ cases
+	{
+		expr:     `{__name__="metric_name"}`,
+		expected: `metric_name`,
+	},
+	{
+		expr:     `{__name__="metric_name",}`,
+		expected: `metric_name`,
+	},
+	{
+		expr:     `{__name__="metric_name", foo="bar"}`,
+		expected: `metric_name{foo="bar"}`,
+	},
+	{
+		expr:     `{__name__="metric_name", foo="bar", first="second"}`,
+		expected: `metric_name{foo="bar", first="second"}`,
+	},
+	{
+		expr:     `{foo="bar", __name__="metric_name"}`,
+		expected: `metric_name{foo="bar",}`,
```

Comment:
IMO there should be space around the `=`.

Reply:
Hm, I guess not. I am yet to see space around `=` anywhere.

Reply:
OK, whatever.

It's not really important to me.

Reply:
That can rather increase the horizontal length by large margins. So, I guess this should be okayish.